test/runner/config.go

```diff
@@ -0,0 +1,326 @@
+package runner
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"github.com/gofrs/flock"
+	"github.com/mitchellh/go-homedir"
+	"github.com/spf13/viper"
+	"gopkg.in/yaml.v1"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"runtime"
+)
+
+const (
+	defaultClusterKey = "default"
+	clustersKey       = "clusters"
+)
+
+var (
+	_, path, _, _     = runtime.Caller(0)
+	certsPath         = filepath.Join(filepath.Dir(filepath.Dir(path)), "certs")
+	deviceConfigsPath = filepath.Join(filepath.Join(filepath.Dir(filepath.Dir(path)), "configs"), "device")
+	storeConfigsPath  = filepath.Join(filepath.Join(filepath.Dir(filepath.Dir(path)), "configs"), "store")
+)
+
+var (
+	configLock *flock.Flock
+)
+
+// getSimulatorPreset gets a device configuration by name
+func getSimulatorPreset(name string) (map[string]interface{}, error) {
+	file, err := os.Open(filepath.Join(deviceConfigsPath, name+".json"))
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	jsonBytes, err := ioutil.ReadAll(file)
+	if err != nil {
+		return nil, err
+	}
+
+	var jsonObj map[string]interface{}
+	err = json.Unmarshal(jsonBytes, &jsonObj)
+	return jsonObj, err
+}
+
+// getSimulatorPresets returns a list of store configurations from the configs/store directory
+func getSimulatorPresets() []string {
+	configs := []string{}
+	filepath.Walk(deviceConfigsPath, func(path string, info os.FileInfo, err error) error {
+		if info.IsDir() {
+			return nil
+		}
+
+		ext := filepath.Ext(info.Name())
+		if ext == ".json" {
+			configs = append(configs, info.Name()[:len(info.Name())-len(ext)])
+		}
```

Comment:
shoudl we drop an error if we don't have `.json` here ?

Reply:
I think we should allow for something like `.md` files to be in the config directories